plugins/org.jkiss.dbeaver.ext.gbase8s/src/org/jkiss/dbeaver/ext/gbase8s/model/meta/GBase8sMetaModel.java

```diff
@@ -323,12 +385,35 @@ JDBCStatement prepareTSObjectLoadStatement(@NotNull JDBCSession session, @NotNul
         }
 
         String catalog = owner.getCatalog() == null ? null : owner.getCatalog().getName();
-        String schemaPattern = owner.getSchema() == null || DBUtils.isVirtualObject(owner.getSchema()) ? null
+        String schema = owner.getSchema() == null || DBUtils.isVirtualObject(owner.getSchema()) ? null
                 : JDBCUtils.escapeWildCards(session, owner.getSchema().getName());
         boolean isOracleMode = GBase8sUtils.isOracleSqlMode(owner.getDataSource().getContainer());
-
-        return session.getMetaData().getTables(isOracleMode ? catalog : schemaPattern,
-                isOracleMode ? schemaPattern : catalog, tableNamePattern, types).getSourceStatement();
+        String ownerPattern = isOracleMode ? schema + "." : catalog + ":";
+        
+        String sql = "SELECT t.tabid, t.tabname AS TABLE_NAME, t.owner AS "
+                + (isOracleMode ? "TABLE_SCHEM" : "TABLE_CATALOG") + ","
+                + " CASE WHEN t.tabtype = 'T' AND t.tabid <= (SELECT tabid FROM systables WHERE trim(tabname) = 'VERSION') THEN 'SYSTEM TABLE'"
+                + " WHEN t.tabtype = 'V' AND t.tabid <= (SELECT tabid FROM systables WHERE trim(tabname) = 'VERSION') THEN 'SYSTEM VIEW'"
+                + " WHEN t.tabtype = 'T' THEN 'TABLE'"
+                + " WHEN t.tabtype = 'V' THEN 'VIEW'"
+                + " ELSE 'SYNONYM'"
+                + " END AS TABLE_TYPE,"
+                + " c.comments AS REMARKS"
+                + " FROM " + ownerPattern + "systables t"
+                + " LEFT JOIN " + ownerPattern + "syscomms c ON t.tabid = c.tabid" + " WHERE t.tabname LIKE ?"
+                + (types != null ? " AND t.tabtype IN (" + String.join(", ", Collections.nCopies(types.length, "?")) + ")" : "");
+        JDBCPreparedStatement dbStat = session.prepareStatement(sql);
+        int paramIndex = 1;
+        dbStat.setString(paramIndex++, tableNamePattern);
+        if (types != null) {
+            for (String type : types) {
+                dbStat.setString(paramIndex++, type);
+            }
+        }
+        return dbStat;
+        // The getTables(...) interface returns a result that does not include information about table comments.
+//        return session.getMetaData().getTables(isOracleMode ? catalog : schemaPattern,
+//                isOracleMode ? schemaPattern : catalog, tableNamePattern, types).getSourceStatement();
```

Comment:
Please remove the commented-out code. This explanation is nice, but rather pointless as custom implementations of meta-models are expected to override the default behavior.